### PR TITLE
feat: add `ao start <url>` one-command project onboarding

### PR DESCRIPTION
**Pull request by AgentWrapper**
## Summary

- Adds URL support to `ao start` — pass a GitHub/GitLab/Bitbucket repo URL to clone, auto-configure, and start the orchestrator in one command
- New `packages/core/src/config-generator.ts` module handles URL parsing, SCM platform detection, project type detection, and config generation
- Auto-detects SCM/tracker plugins from URL host, default branch from local refs, language/package manager from project files
- Reuses existing clones and configs when found — idempotent operation

### Usage
```bash
ao start https://github.com/ComposioHQ/DevOS
```

This will:
1. Clone the repo (or reuse existing)
2. Generate `agent-orchestrator.yaml` with smart defaults
3. Start dashboard + orchestrator

### URL formats supported
- `https://github.com/owner/repo`
- `https://github.com/owner/repo.git`
- `git@github.com:owner/repo.git`
- Same patterns for GitLab and Bitbucket

### Edge cases handled
- Repo already cloned in CWD → reuses it
- Config already exists → uses as-is
- SSH URLs → converted to HTTPS for cloning
- Unknown SCM hosts → works but skips SCM/tracker plugin config

## Test plan
- [x] 36 unit tests covering URL parsing, SCM detection, project detection, config generation, clone target resolution
- [x] `pnpm typecheck` passes (0 errors)
- [x] `pnpm lint` passes (0 errors, only pre-existing warnings)
- [x] `pnpm test` passes (all suites)
- [ ] Manual test: `ao start https://github.com/ComposioHQ/DevOS`
- [ ] Manual test: existing `ao start <project>` still works unchanged

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/cli/src/commands/start.ts [hunk 1]
```diff
@@ -1,29 +1,60 @@
...
+    const free = await isPortAvailable(port);
+    if (!free) {
+      // Port is listening — server is ready
+      const browser = spawn("open", [url], { stdio: "ignore" });
+      browser.on("error", () => {});
+      return;
+    }
+    await new Promise((r) => setTimeout(r, 300));
+  }
+}
```
[the remark this answers is not in the document]
Comment (AgentWrapper) [reply packages/cli/src/commands/start.ts]: Fixed in 324c28d — moved `waitForPortAndOpen` to `lib/web-dir.ts` as a shared utility. Both `start.ts` and `dashboard.ts` now import the same cancellation-aware implementation with `AbortSignal` support. `start.ts` now properly aborts polling on early dashboard exit.

File: packages/cli/src/commands/start.ts [hunk 2]
```diff
@@ -134,109 +333,22 @@ export function registerStart(program: Command): void {
...
-            console.log(chalk.cyan("Orchestrator:"), `already running (${sessionId})`);
+          let config: OrchestratorConfig;
+          let projectId: string;
+          let project: ProjectConfig;
+
+          // Detect URL argument — run onboarding flow
+          if (projectArg && isRepoUrl(projectArg)) {
+            console.log(chalk.bold.cyan("\n  Agent Orchestrator — Quick Start\n"));
+            config = await handleUrlStart(projectArg);
+            ({ projectId, project } = resolveProject(config));
```
[the remark this answers is not in the document]
Comment (AgentWrapper) [reply packages/cli/src/commands/start.ts:344]: Fixed in PR #272 — `handleUrlStart` now returns parsed URL info, and `resolveProjectByRepo()` matches by `owner/repo` against config projects before falling back to standard resolution.

File: packages/cli/src/lib/web-dir.ts [hunk 3]
```diff
@@ -36,6 +37,29 @@ export function isPortAvailable(port: number): Promise<boolean> {
...
+  url: string,
+  signal: AbortSignal,
+  timeoutMs = 30_000,
+): Promise<void> {
+  const start = Date.now();
+  while (!signal.aborted && Date.now() - start < timeoutMs) {
+    const free = await isPortAvailable(port);
+    if (!free) {
+      const browser = spawn("open", [url], { stdio: "ignore" });
+      browser.on("error", () => {});
```
[the remark this answers is not in the document]
Comment (AgentWrapper) [reply packages/cli/src/lib/web-dir.ts:56]: Fixed in PR #272 — now uses `spawn("cmd.exe", ["/c", "start", "", url])` on Windows, `xdg-open` on Linux, and `open` on macOS.

**Comment by jleechan2015** [comment]
**🤖 Skeptic Agent Verdict (bd-qw6)**

VERDICT: SKIPPED

_Posted by jleechan2015 · 2026-03-29T07:24:56.936Z_

**Comment by jleechan2015** [comment]
**🤖 Skeptic Agent Verdict (bd-qw6)**

VERDICT: SKIPPED

_Posted by jleechan2015 · 2026-03-29T09:10:24.109Z_
